### PR TITLE
Use Kubernetes APIs to construct user kubeconfigs

### PR DESCRIPTION
**Pull request by twz123**
## Description

Replace the Go template code with the typed structs and APIs provided by upstream Kubernetes. This avoids manual cert loading and provides type safety.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [ ] Manual test
- [ ] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: cmd/kubeconfig/create.go [hunk 1]
```diff
@@ -89,54 +66,57 @@ Note: A certificate once signed cannot be revoked for a particular user`,
...
+	userCert, err := certManager.EnsureCertificate(userReq, "root")
+	if err != nil {
+		return nil, fmt.Errorf("failed generate user certificate: %w, check if the control plane is initialized on this node", err)
+	}
+
+	const k0sContextName = "k0s"
+	kubeconfig := clientcmdapi.Config{
+		Clusters: map[string]*clientcmdapi.Cluster{k0sContextName: {
+			Server:               clusterAPIURL,
+			CertificateAuthority: userReq.CACert,
```
Comment (jnummelin) [inline cmd/kubeconfig/create.go:105]: hmm, doesn't this just make it a path to the file? I think we'd want to use `CertificateAuthorityData` instead

**Review by jnummelin — COMMENTED** [review]
I wonder if we should have some e2e test covering this too. I mean some test that would create new kubeconfig for a user and verify that it actually works

File: cmd/kubeconfig/create.go (hunk 1 above)
Comment (jnummelin) [reply cmd/kubeconfig/create.go:105]: ah, right.

**Comment by twz123** [comment]
> I wonder if we should have some e2e test covering this too. I mean some test that would create new kubeconfig for a user and verify that it actually works

I think we can get away with unit tests here.  I've added some asserts to make sure the certs are properly embedded in 41cb3e6471638e60e956ef76b2efa02611657cc9.

I've scanned the existing inttests, and there's indeed the PSP inttest which invokes `k0s kubeconfig create test` and uses the generated kubeconfig to connect to the cluster. We need to do something about PSPs _eventually_ anyways, as noted in #1420. I reckon we want to add some real RBAC tests when doing the migration away from PSPs, then.

**Review by jnummelin — APPROVED** [review]

**Pull request merged**